dynamic_rest/paginator.py

```diff
@@ -0,0 +1,96 @@
+# adapter from Django's django.core.paginator
+# adds support for the "exclude_count" parameter
+
+from math import ceil
+
+import inspect
+from django.utils.functional import cached_property
+from django.core.paginator import Paginator
+from django.utils.inspect import method_has_no_args
+
+try:
+    from django.utils.translation import gettext_lazy as _
+except ImportError:
+    def _(x):
+        return x
+
+
+class InvalidPage(Exception):
+    pass
+
+
+class PageNotAnInteger(InvalidPage):
+    pass
+
+
+class EmptyPage(InvalidPage):
+    pass
+
+
+class DynamicPaginator(Paginator):
+
+    def __init__(self, object_list, per_page, orphans=0,
+                 allow_empty_first_page=True, exclude_count=False):
+        self.object_list = object_list
+        self._check_object_list_is_ordered()
+        self.per_page = int(per_page)
+        self.orphans = int(orphans)
+        self.allow_empty_first_page = allow_empty_first_page
+        self.exclude_count = exclude_count
+
+    def validate_number(self, number):
```

Comment:
what version of django did you use to get these method's django defn? maybe we should add that info here as a comment?

Reply:
This was taken from 3.1, but it seems to work across all tested versions in DREST tox (from 2.2 onwards)

Reply:
done